nemo/collections/asr/models/classification_models.py

```diff
@@ -471,113 +652,149 @@ def change_labels(self, new_labels: List[str]):
 
             logging.info(f"Changed decoder output to {self.decoder.num_classes} labels.")
 
-    def _update_decoder_config(self, cfg):
+    def _update_decoder_config(self, labels, cfg):
         """
         Update the number of classes in the decoder based on labels provided.
 
         Args:
+            labels: The current labels of the model
             cfg: The config of the decoder which will be updated.
         """
         OmegaConf.set_struct(cfg, False)
 
-        labels = self.cfg.labels
-
         if 'params' in cfg:
             cfg.params.num_classes = len(labels)
         else:
             cfg.num_classes = len(labels)
 
         OmegaConf.set_struct(cfg, True)
 
-    def _setup_transcribe_dataloader(self, config: Dict) -> 'torch.utils.data.DataLoader':
+
+class EncDecRegressionModel(_EncDecBaseModel):
+    """Encoder decoder class for speech regression models.
+    Model class creates training, validation methods for setting up data
+    performing model forward pass.
+    """
+
+    @classmethod
+    def list_available_models(cls) -> List[PretrainedModelInfo]:
         """
-        Setup function for a temporary data loader which wraps the provided audio file.
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
+        Returns:
+            List of available pre-trained models.
+        """
+        result = []
+
+        return result
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        if not cfg.is_regression_task:
```

Comment:
Do `if 'is_regression_task' in cfg and not cfg.is_regression_task:` to first check if `is_regression_task` exists or not.

Reply:
Same as above

Reply:
Yes my bad, it should mandatorily check that 1) cfg.is_regression_task exists, and if it does exist, it is set to True. If either condition is not met, raise an error in the regression model.